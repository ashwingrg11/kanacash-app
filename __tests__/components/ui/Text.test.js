import 'react-native';
import React from 'react';

import {
  RegularText,
  MediumText,
  SemiBoldText,
  BoldText,
  LightText,
} from '~/components/ui/Text';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<Text/>', () => {
  it('RegularText renders correctly', () => {
    const tree = renderer.create(<RegularText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('MediumText renders correctly', () => {
    const tree = renderer.create(<MediumText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SemiBoldText renders correctly', () => {
    const tree = renderer.create(<SemiBoldText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('BoldText renders correctly', () => {
    const tree = renderer.create(<BoldText />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('LightText renders correctly', () => {
    const tree = renderer.create(<LightText />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
