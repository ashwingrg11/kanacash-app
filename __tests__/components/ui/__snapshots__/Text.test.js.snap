// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> BoldText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "OpenSans-Bold",
        "fontSize": 24,
        "fontWeight": "600",
        "lineHeight": 41,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Text/> LightText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#5C5C5C",
        "fontFamily": "OpenSans-Light",
        "fontSize": undefined,
        "lineHeight": 22,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Text/> MediumText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1F1F1F",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 18,
        "fontWeight": "500",
        "lineHeight": 22,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Text/> RegularText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#5C5C5C",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
        "lineHeight": 20,
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`<Text/> SemiBoldText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "OpenSans-SemiBold",
        "fontSize": 20,
        "lineHeight": 25,
      },
      undefined,
      undefined,
    ]
  }
/>
`;
